UniversalAlgebra/Terms/Basic.lagda

```diff
@@ -63,9 +67,46 @@ The definition of `Term X` is recursive, indicating that an inductive type could
 
 data Term (X : Type χ ) : Type (ov χ)  where
  ℊ : X → Term X    -- (ℊ for "generator")
- node : (f : ∣ 𝑆 ∣)(𝑡 : ∥ 𝑆 ∥ f → Term X) → Term X
+ node : (f : ∣ 𝑆 ∣)(t : ∥ 𝑆 ∥ f → Term X) → Term X
 
-open Term public
+module _ {X : Type χ } where
+
+
+ data _≐_ : Term X → Term X → Type (𝓞 ⊔ 𝓥 ⊔ lsuc χ) where
```

Comment:
This is indeed a much better encoding of `_≐_` .  You get injectivity, pattern-matching and no-bad-branches for free.  Proof-relevance FTW!